File: app/src/main/java/me/imzack/app/ender/injector/component/MyPlansComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.MyPlansPresenterModule;
import me.imzack.app.ender.injector.scope.FragmentScope;
import me.imzack.app.ender.view.fragment.MyPlansFragment;

import dagger.Component;

@FragmentScope
@Component(dependencies = AppComponent.class, modules = MyPlansPresenterModule.class)
public interface MyPlansComponent {

    void inject(MyPlansFragment myPlansFragment);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/MyPlansComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.MyPlansPresenterModule
import net.zackzhang.app.end.injector.scope.FragmentScope
import net.zackzhang.app.end.view.fragment.MyPlansFragment

@FragmentScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(MyPlansPresenterModule::class))
interface MyPlansComponent {

    fun inject(myPlansFragment: MyPlansFragment)
}
